ra2ce/analysis/losses/risk_calculation/risk_calculation_base.py

```diff
@@ -6,13 +6,30 @@
 
 
 class RiskCalculationBase(ABC):
+    risk_calculation_year: int
+    losses_gdf: gpd.GeoDataFrame
+
     def __init__(self, risk_calculation_year: int, losses_gdf: gpd.GeoDataFrame):
-        self.risk_calculation_year: int = risk_calculation_year
-        self.losses_gdf: gpd.GeoDataFrame = losses_gdf
-        self.return_periods: set = self._get_return_periods()
-        self.max_return_period: int = max(self.return_periods)
-        self.min_return_period: int = min(self.return_periods)
-        self._to_integrate: gpd.GeoDataFrame = self._get_to_integrate()
+        self.risk_calculation_year = risk_calculation_year
+        self.losses_gdf = losses_gdf
+        self.__post_init__()
+
+    def __post_init__(self):
+        """Private method to __post_init__ computed attributes."""
+        self._to_integrate = self._get_to_integrate()
+        self._rework_damage_data()
+
+    @property
+    def return_periods(self) -> set[int]:
+        return self._get_return_periods()
+
+    @property
+    def max_return_period(self) -> int:
+        return max(self.return_periods)
+
+    @property
+    def min_return_period(self) -> int:
```

Comment:
If they are only being used by inheriting classes you should mark them as protected.
```suggestion
    def _return_periods(self) -> set[int]:
        return self._get_return_periods()

    @property
    def _max_return_period(self) -> int:
        return max(self.return_periods)

    @property
    def _min_return_period(self) -> int:
```